src/Compilers/Core/Portable/InternalUtilities/HashAlgorithms.cs

```diff
@@ -90,19 +117,38 @@ protected HashAlgorithm(IDisposable hashInstance)
 
         public byte[] ComputeHash(byte[] bytes)
         {
-            return (byte[])s_bytesMethod.Invoke(_hashInstance, new object[] { bytes });
+            return (byte[])s_ComputeHash_bytes_Method.Invoke(_hashInstance, new object[] { bytes });
         }
 
         public byte[] ComputeHash(byte[] bytes, int offset, int count)
         {
-            return (byte[])s_bytesOffsetCountMethod.Invoke(_hashInstance, new object[] { bytes, offset, count });
+            return (byte[])s_ComputeHash_bytesOffsetCount_Method.Invoke(_hashInstance, new object[] { bytes, offset, count });
         }
 
         public byte[] ComputeHash(Stream stream)
         {
-            return (byte[])s_streamMethod.Invoke(_hashInstance, new object[] { stream });
+            return (byte[])s_ComputeHash_stream_Method.Invoke(_hashInstance, new object[] { stream });
+        }
+
+        public void TransformBlock(byte[] inputBuffer, int inputCount)
```

Comment:
I don't think I can. It just forwards to the TransformBlock method of the platform HashAlgorithm API. The documentation is linked on line 54. The only reason I have a loop is that the documentation for the HashAlgorithm API is unclear whether or not the caller requires one (that is whether it always transforms all of the provided data).